runtime/calamari/src/lib.rs

```diff
@@ -975,6 +1001,21 @@ impl_runtime_apis! {
         }
     }
 
+    impl nimbus_primitives::NimbusApi<Block> for Runtime {
+        fn can_author(author: NimbusId, relay_parent: u32, parent_header: &<Block as BlockT>::Header) -> bool{
+            System::initialize(&(parent_header.number + 1), &parent_header.hash(), &parent_header.digest);
+
+            // And now the actual prediction call
+            <AuthorInherent as nimbus_primitives::CanAuthor<_>>::can_author(&author, &relay_parent)		}
+    }
+
+    // We also implement the old AuthorFilterAPI to meet the trait bounds on the client side.
+    impl nimbus_primitives::AuthorFilterAPI<Block, NimbusId> for Runtime {
+        fn can_author(_: NimbusId, _: u32, _: &<Block as BlockT>::Header) -> bool {
+            panic!("AuthorFilterAPI is no longer supported. Please update your client.")
```

Comment:
Is anywhere else calling this API? I mean `panic` will cause runtime crash. Not sure it's safe.

Reply:
Yes, it's meant to crash the runtime because that old API is no longer supported.
We have to provide it because upstream nimbus still has it for backwards compatibility, but doesn't use it anywhere anymore.
Our client also never used this, so we don't care.